rfc/rfc023-x509credential.md

```diff
@@ -0,0 +1,504 @@
+
+- If weak cryptographic algorithms (e.g., MD5, SHA-1) or small key sizes (e.g., <2048-bit RSA) are used, the
+  certificates or their signatures could be cracked by modern computational power.
+- **Consequences**:
+  - An attacker could forge or spoof certificates.
+  - Sensitive data could be decrypted easily, exposing confidential information such as passwords, personal data, etc.
+
+### **5. Improper Credential Subject Validation**
+
+- If the `credentialSubject` field in frameworks like `X509Credential` is not properly validated, it may allow
+  fields not aligned with the X.509 certificate to be added or accepted.
+- **Consequences**:
+  - Attackers could inject unauthorized or false data (e.g., incorrect organization name or purpose), tricking verifiers
+    by impersonating trusted entities.
+  - Loss of trust in the system due to inconsistencies between certificates and credentials.
+
+### **6. Forged Proofs or Tampered Credentials**
+
+- Failure to verify cryptographic proofs tied to certificates could allow credentials or data to be tampered with.
+- **Consequences**:
+  - Credentials could be modified to grant unauthorized access.
+  - The integrity of systems relying on these credentials could be compromised.
+
+### **7. Missing Root CA Verification**
+
+- If the source of trust (Root CA) is not explicitly verified and trusted, attackers could use certificates issued by
+  unapproved CAs.For instance, in case of UZI certificates the `ca-fingerprint` must match the hash of either the Root
+  CA or one of the Intermediate CAs as published by the [UZI register](https://www.zorgcsp.nl/ca-certificaten).
+- **Consequences**:
+  - Fake certificates signed by rogue or unvalidated CAs could be accepted as valid.
+  - Attackers gain the ability to impersonate legitimate entities in scenarios such as encrypted communication or
+    identity verification.
+
+### **8. Certificate Misuse**
+
+- Without proper validation of certificate attributes (e.g., URA number in UZI certificates), certificates may be
+  misused in unintended contexts.
+- **Consequences**:
+  - Fraud or impersonation using certificates outside their intended scope.
+  - Misrepresentation of organizations or individuals.
+
+### **9. Lack of Reliable Revocation Handling**
+
+- If revocation checks poorly handle network issues or failures (e.g., OCSP response unavailability), it could result in
+  the acceptance of revoked or invalid certificates.
+- **Consequences**:
+  - Increased risk of improper trust, allowing revoked credentials to function within the system.
+  - Security-critical applications become susceptible to breaches.
+
+## References
+
+- [Verifiable Credentials Data Model v1.1](https://www.w3.org/TR/vc-data-model/)
+- [DID:X509 Method Specification](https://trustoverip.github.io/tswg-did-x509-method-specification/)
+- [X.509 Certificate Revocation (OCSP/CRL)](https://datatracker.ietf.org/doc/html/rfc5280)
+
+# An application of the RFC023: UZI server certificates
```

Comment:
Propose to move this to the wiki. Future changes will see UZI being removed in favour of PKIo.

Reply:
Moved to https://wiki.nuts.nl/books/x509credential